src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -1216,10 +1216,16 @@ Namespace Microsoft.CodeAnalysis.Semantics
             Dim syntax As SyntaxNode = boundReturnStatement.Syntax
             Dim type As ITypeSymbol = Nothing
             Dim constantValue As [Optional](Of Object) = New [Optional](Of Object)()
-            Dim isImplicit As Boolean = boundReturnStatement.WasCompilerGenerated
+            Dim isImplicit As Boolean = boundReturnStatement.WasCompilerGenerated OrElse IsEndSubOrFunctionStatement(syntax)
```

Comment:
When is `boundReturnStatement.WasCompilerGenerated = False` and `IsEndSubOrFunctionStatement(syntax) = True`?

Reply:
> When is boundReturnStatement.WasCompilerGenerated = False and IsEndSubOrFunctionStatement(syntax) = True?

See unit-test IBlockStatement_SubMethodBlock for example.

Reply:
Ah, ignore my previous comment, I see you are handling it here.